### DISCUSSION TITLE
How to regenerate build tree for source code debugging?

### DISCUSSION THREAD

**Discussion by DownerCase** [microsoft/vcpkg#33259] [Q&A]
I feel like I've missed something obvious but humour me...

Suppose I have built a library, libfoo, which is quite large and have since cleaned out my `buildtrees` and `downloads` folder to save space. I have now found an issue with libfoo that needs debugging, so I start up my debugging session and... I have no source files to debug against. 😢 

How do I politely ask vcpkg to put the source files back (preferably with the patches applied)?
When I had this problem recently I:

- Went into the ports folder and found libfoo
- Copied the source code URL from the portfile
- Cloned it somewhere for safekeeping
- Told my debugger where to look

Not very convenient.

Ideally I would be able to temporarily write in my manifest that I want to have the source tree for my dependency (and all of *its* dependencies) available regardless of any other factors (eg: it having been previously deleted, built on a different machine, or any "clean after build" instructions).

If this is something genuinely missing and would be a good feature, I'd potentially be interested in giving it a go.

**Comment by Osyotr** [comment]
`vcpkg install --only-downloads` will download and unpack sources for you. Works in manifest mode, too.

**Reply by DownerCase** [reply]
Ahh thanks, I’ll give it a go. Not ideal to pull everything but better than nothing I guess.
